src/views/MassEditor.vue

```
@@ -0,0 +1,311 @@
+      const categories = [
+        { name: this.$t('basic'), fields: ['Name', 'SteamLogin', 'SteamPassword', 'Enabled', 'Paused', 'OnlineStatus', 'BotBehaviour'] },
+        { name: this.$t('security'), fields: ['PasswordFormat', 'UseLoginKeys'] },
+        { name: this.$t('access'), fields: ['SteamUserPermissions', 'SteamParentalCode'] },
+        { name: this.$t('trade'), fields: ['SteamTradeToken', 'AcceptGifts', 'SendTradePeriod', 'SendOnFarmingFinished', 'CompleteTypesToSend', 'TradingPreferences', 'LootableTypes', 'TransferableTypes', 'MatchableTypes'] },
+        { name: this.$t('farming'), fields: ['FarmingOrders', 'AutoSteamSaleEvent', 'FarmPriorityQueueOnly', 'FarmNonRefundableGamesOnly', 'FarmOffline', 'ShutdownOnFarmingFinished'] },
+        { name: this.$t('customization'), fields: ['SteamMasterClanID', 'UserInterfaceMode', 'RedeemingPreferences', 'GamesPlayedWhileIdle', 'CustomGamePlayedWhileFarming', 'CustomGamePlayedWhileIdle'] },
+        { name: this.$t('performance'), fields: ['HoursUntilCardDrops'] },
+      ];
+
+      return {
+        loading: true,
+        saving: false,
+        fields: [],
+        model: {},
+        descriptions: {},
+        categories,
+        config: {},
+        status: 'bots',
+        selectedBots: [],
+        selectedConfigProperties: [],
+        placeholder: this.$t('mass-editor-properties-placeholder'),
+        watcherActive: true,
+      };
+    },
+    computed: {
+      ...mapGetters({
+        version: 'asf/version',
+        displayCategories: 'settings/displayCategories',
+        bots: 'bots/bots',
+      }),
+      configProperties() {
+        return this.fields.filter(field => field.param);
+      },
+      configOptions() {
+        return this.fields.map(field => field.param);
+      },
+      selectedBotNames() {
+        return this.selectedBots.map(bot => bot.name);
+      },
+    },
+    watch: {
+      async bots() {
+        // Needed because 'this.bots[0].name' is sometimes undefined
```

What exactly is _needed_ here? The `loadBotConfig` function is called and should throw an error, when `bots` array is empty. I do not see how the `watcherActive`, initially set to `true`, prevents this.

---

I have no idea but it does _some kind of magic_

Maybe something with the lifecycle? Watcher running after store is inizialized? I have no idea. But now that I refactored it to be its own component it _should_ be fixed and we do not need this. I will try it and report back.

---

Looks like it is fixed now. 👍